src/yara.cc

```diff
@@ -952,19 +953,118 @@ int scanCallback(int message, void* data, void* param) {
 	return CALLBACK_CONTINUE;
 }
 
+NAN_METHOD(ScannerWrap::ReconfigureRules) {
+	Nan::HandleScope scope;
+
+	ScannerWrap* scanner = ScannerWrap::Unwrap<ScannerWrap>(info.This());
+
+	scanner->lock_read();
+
+	bool rules_compiled = scanner->rules ? true : false;
+
+	if (! rules_compiled) {
+		Nan::ThrowError("Please call configure() before reconfigureRules()");
+		return;
+	}
+	Local<Object> options = Nan::To<Object>(info[0]).ToLocalChecked();
+
+	Local<Array> variables = Local<Array>::Cast(
+			Nan::Get(options, Nan::New("variables").ToLocalChecked()).ToLocalChecked()
+		);
+
+	for (uint32_t i = 0; i < variables->Length(); i++) {
+		if (Nan::Get(variables, i).ToLocalChecked()->IsObject()) {
+			Local<Object> variable = Nan::To<Object>(Nan::Get(variables, i).ToLocalChecked()).ToLocalChecked();
+
+			VarType type;
+			std::string id;
+
+			Local<Uint32> t = Nan::To<Uint32>(Nan::Get(variable, Nan::New("type").ToLocalChecked()).ToLocalChecked()).ToLocalChecked();
+			type = (VarType) t->Value();
+
+			Local<String> i = Nan::To<String>(Nan::Get(variable, Nan::New("id").ToLocalChecked()).ToLocalChecked()).ToLocalChecked();
+			id = *Nan::Utf8String(i);
```

Comment:
A few comments here.

Is the cast really necessary? If so, would it be better to use one of the more type safe casts, like `std::static_cast` or `std::dynamic_cast`?

Also, it's usually considered better to initialize variables on construction in C++. Saves you from first constructing an object, and then later calling the assignment operator.

It may also be a good idea to use type inference. It will most likely remove the need for the cast.

```c++
auto type = t->value();
...
auto is = *Nan::Utf8String(i);
```

Reply:
I didn't know C++ had auto! Thanks for the tip! I tried playing around with it last night but it always gave me invalid conversion unless I left the cast there.

Reply:
`auto` is new since my C++ time too, so I can't help you there :-)

Reply:
C++ has improved leaps and bounds since 2011! `auto` was one of the first immediate improvements of C++11, but there's a lot of good stuff in the newer standards. Properly written C++ today is both efficient, type safe, memory safe and quite elegant compared to the pre 2011 variant.

But leave the cast for now, we can have a go at it later if you want. Could be educational for everybody!